lib/smart_name/contextual.rb

```diff
@@ -0,0 +1,120 @@
+class SmartName
+  module Contextual
```

Comment:
contextual names being pretty important, better docs on these ideas would also be really helpful. It is pretty cool that this logic is just depending on names, and not the underlying data that they index (cards).